apiextensions/v1alpha1/types_compatibilityrequirement.go

```diff
@@ -0,0 +1,368 @@
+	// reconciliation cannot progress due to an invalid spec. The controller
+	// will not reconcile this object again until the spec is updated.
+	CompatibilityRequirementConfigurationErrorReason string = "ConfigurationError"
+
+	// CompatibilityRequirementTransientErrorReason indicates that
+	// reconciliation failed due to an error that can be retried.
+	CompatibilityRequirementTransientErrorReason string = "TransientError"
+
+	// CompatibilityRequirementUpToDateReason surfaces when reconciliation
+	// completed successfully for the condition's observed generation.
+	CompatibilityRequirementUpToDateReason string = "UpToDate"
+)
+
+// CompatibilityRequirement's Admitted condition and corresponding reasons.
+const (
+	// CompatibilityRequirementAdmitted is true if the requirement has been configured in the validating webhook,
+	// otherwise false.
+	CompatibilityRequirementAdmitted string = "Admitted"
+
+	// CompatibilityRequirementAdmittedReason surfaces when the requirement has been configured in the validating webhook.
+	CompatibilityRequirementAdmittedReason string = "Admitted"
+
+	// CompatibilityRequirementNotAdmittedReason surfaces when the requirement has not been configured in the validating webhook.
+	CompatibilityRequirementNotAdmittedReason string = "NotAdmitted"
+)
+
+// CompatibilityRequirement's Compatible condition and corresponding reasons.
+const (
+	// CompatibilityRequirementCompatible is true if the observed CRD is compatible with the requirement,
+	// otherwise false. Note that Compatible may be false when adding a new requirement which the existing
+	// CRD does not meet.
+	CompatibilityRequirementCompatible string = "Compatible"
+
+	// CompatibilityRequirementRequirementsNotMetReason surfaces when a CRD exists, and it is not compatible with this requirement.
+	CompatibilityRequirementRequirementsNotMetReason string = "RequirementsNotMet"
+
+	// CompatibilityRequirementCRDNotFoundReason surfaces when the referenced CRD does not exist.
+	CompatibilityRequirementCRDNotFoundReason string = "CRDNotFound"
+
+	// CompatibilityRequirementCompatibleWithWarningsReason surfaces when the CRD exists and is compatible with this requirement, but Message contains one or more warning messages.
+	CompatibilityRequirementCompatibleWithWarningsReason string = "CompatibleWithWarnings"
+
+	// CompatibilityRequirementCompatibleReason surfaces when the CRD exists and is compatible with this requirement.
+	CompatibilityRequirementCompatibleReason string = "Compatible"
+)
+
+// CompatibilityRequirementStatus defines the observed status of the Compatibility Requirement.
+// +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.crdName) || has(self.crdName) && oldSelf.crdName == self.crdName",message="crdName cannot be changed once set"
+type CompatibilityRequirementStatus struct {
+	// conditions is a list of conditions and their status.
+	// Known condition types are Progressing, Admitted, and Compatible.
```

Comment:
Do we know enough about these conditions to be able to document what they mean here?

Reply:
Added PTAL